src/IO/configuration.cpp

```diff
@@ -360,4 +361,113 @@ namespace config {
 
 }
 
+/*!
+    \class ConfigStringParser
```

Comment:
Is this new class really needed? Can we just add an overload to the read/write functions of ConfigParser, this overloads will take in input the string and the format?